internal/internal_event_handlers.go

```diff
@@ -818,3 +858,30 @@ func (weh *workflowExecutionEventHandlerImpl) handleChildWorkflowExecutionTermin
 
 	return nil
 }
+
+func (weh *workflowExecutionEventHandlerImpl) handleSignalExternalWorkflowExecutionRequested(event *m.HistoryEvent) error {
+	attributes := event.ExternalWorkflowExecutionSignalRequestedEventAttributes
+	signalID := string(attributes.Control)
+	decision := weh.decisionsHelper.handleSignalExternalWorkflowExecutionRequested(signalID)
+	signal := decision.getData().(*scheduledSignal)
+	if signal.handled {
+		return nil
+	}
+	signal.handle(nil, nil)
+
+	return nil
+}
+
+func (weh *workflowExecutionEventHandlerImpl) handleSignalExternalWorkflowExecutionFailed(event *m.HistoryEvent) error {
+	attributes := event.SignalExternalWorkflowExecutionFailedEventAttributes
+	signalID := string(attributes.Control)
+	decision := weh.decisionsHelper.handleSignalExternalWorkflowExecutionFailed(signalID)
+	signal := decision.getData().(*scheduledSignal)
+	if signal.handled {
+		return nil
+	}
+	err := fmt.Errorf("signal external workflow failed, %v", attributes.GetCause())
```

Comment:
probably better to define a new error type for this.

Reply:
I will keep simple like this, to be consistent with Child, Timer and others.

Reply:
We, probably have to go over errors and see if there are any actionable ones like: already running for child or workflow doesn't exist for send signal.

Reply:
Add issue: https://github.com/uber-go/cadence-client/issues/345